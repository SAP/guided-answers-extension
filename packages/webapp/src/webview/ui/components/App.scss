html {
    height: 100%;
}
* {
    font-family: var(--vscode-font-family);
    -webkit-font-smoothing: antialiased;
}

h1 {
    font-weight: 300;
    color: var(--vscode-foreground);
    font-size: 24px;
}

h2 {
    font-size: 13px;
}

.guided-answer {
    &__header {
        overflow: hidden;
        position: fixed; /* Set the navbar to fixed position */
        top: 0; /* Position the navbar at the top of the page */
        width: 100%;
        z-index: 10;
        background-color: var(--vscode-editor-background);
        &__title {
            font-weight: 600;
            margin: 0px;
            color: var(--vscode-settings-headerForeground);
            line-height: 29px;
        }
        &__subtitle {
            color: var(--vscode-editorCodeLens-foreground, #999);
            font-size: 2em;
        }
        &__goback {
            height: 26px;
            margin-top: 17px;
        }
        &__logoAndTitle {
            display: flex;
            align-items: center;
            margin-top: 40px;
        }
        &__searchField {
            display: flex;
            align-items: center;
            flex-direction: column;
            margin-top: -28px;
            margin-bottom: 40px;
        }
    }
    &__container {
        margin-top: 111px;
        padding-bottom: 10px;
    }
    &__tree {
        width: 100%;
        height: auto;
        cursor: pointer;
        text-align: left;
        display: flex;
        border: none;
        &:hover {
            background-color: var(--vscode-list-hoverBackground);
        }

        &__ul {
            padding: 15px;
            &h3 {
                margin-bottom: 10px;
            }
        }
        &__title {
            font-size: 14px;
            color: var(--vscode-settings-headerForeground);
            margin-bottom: 10px;
            margin-top: 0px;
        }
        &__desc {
            color: var(--vscode-foreground);
            height: auto;
            font-size: 13px;
        }
    }
}

.tree-search-field {
    width: 33%;
    min-width: 250px;
}

.tree-search-icon {
    margin-top: 5px;
    text-align: center;
}

.tree-item {
    margin: 0;
}

#sap-logo {
    width: 59px;
}

.striped-list {
    padding-left: 0px;
    margin-top: 0px;
}

ul.striped-list > li:nth-of-type(odd) {
    background-color: var(--vscode-breadcrumbPicker-background);
    padding-left: 0px;
}

li {
    background-color: var(--vscode-editor-background);
    list-style-type: none;
}
