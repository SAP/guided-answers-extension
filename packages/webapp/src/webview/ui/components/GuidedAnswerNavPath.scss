* {
    box-sizing: border-box;
    outline: none;
}

h1 span {
    font-weight: 600;
}

.container {
    width: 100%;
    margin-top: 107px;
    position: relative;
    overflow: hidden;
    padding-left: 20px;
    padding-right: 20px;
}

.vertical-line {
    display: block;
    left: 9px;
    width: 1px;
    position: absolute;
    top: 47px;
    margin-left: -1px;
    height: 55px;
    background: lightgray;
    z-index: 1;
}

.timeline-block {
    width: 100%;
    margin: 0px 0px 0px 10px;
    display: flex;
    justify-content: space-between;
    clear: both;
    position: relative;
    float: none;
    height: 80px;
}

.marker {
    position: absolute;
    top: 20px;
    font-size: 20px;
}

.timeline-content {
    width: 100%;
    margin: 10px 10px 10px 35px;
    border: 1px solid lightgray;
    color: #666;
}

.timeline-content-arrow {
    position: absolute;
    top: 16px;
    left: -7px;
    position: relative;
    width: 0.75rem;
    height: 0.75rem;
    background-color: var(--vscode-editor-background);
    opacity: 1;
    -moz-opacity: 1;
    -webkit-opacity: 1;
    border-bottom: 0.0625rem solid #d1d1d1;
    border-left: 0.0625rem solid #d1d1d1;
    transform: translateX(0px) translateY(0px) rotate(45deg);
    -moz-transform: translateX(0px) translateY(0px) rotate(45deg);
    z-index: 1;
}

.timeline__path {
    display: inline-block;
    color: var(--vscode-textLink-foreground);
    cursor: pointer;
    padding: 1px 7px 2px 18px;
}
