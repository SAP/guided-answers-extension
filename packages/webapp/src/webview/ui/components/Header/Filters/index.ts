export * from './Filters';
export * from './FiltersRibbon';
