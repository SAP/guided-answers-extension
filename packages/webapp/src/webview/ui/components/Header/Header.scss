.guided-answer {
    &__header {
        display: flex;
        width: 100%;
        z-index: 10;
        background-color: var(--vscode-editor-background);
        position: sticky;
        top: 0;
        padding-bottom: 20px;
        &__sub {
            display: flex;
            align-items: center;
            width: 100%;
        }
        &__title {
            font-weight: 600;
            margin: 0;
            color: var(--vscode-settings-headerForeground);
            line-height: 29px;
        }
        &__subtitle {
            color: var(--vscode-editorCodeLens-foreground, #999);
            font-size: 2em;
        }
        &__navButtons {
            height: 25px;
            margin-top: 30px;
            color: var(--vscode-foreground);
            padding: 0 5px 0px 5px;
            text-align: start;
            cursor: pointer;
            background: none;
            border: 0;
            &__content {
                vertical-align: middle;
                display: inline-block;
                &__text {
                    margin-left: 3px;
                }
            }
            display: flex;
            align-items: center;
            svg {
                width: 16px;
                height: 16px;
            }
        }
        &__navButtons:hover {
            background: var(--vscode-menubar-selectionBackground);
            outline: 1px solid var(--vscode-contrastActiveBorder);
            border-radius: 3px;
            > span {
                text-decoration: none;
            }
        }
        &__navButtons:focus-visible {
            outline: 1px solid var(--vscode-focusBorder, #007fd4);
        }

        &__allAnswersButton {
            width: 25%;
            margin-right: 80px;
        }
        &__back-restart-buttons {
            display: flex;
            justify-content: space-between;
            width: 75%;
            max-width: 620px;
        }
        &__restart-feedback-buttons {
            display: flex;
        }
        &__divider {
            border-left: 1px solid var(--vscode-editorWidget-border);
            height: 16px;
            margin-top: 35px;
            margin-left: 10px;
            margin-right: 10px;
        }
        &__logoAndTitle {
            display: flex;
            align-items: center;
            margin-top: 30px;
            min-width: 280px;
        }
        &__searchField {
            display: flex;
            align-items: start;
            flex-direction: row;
            margin-top: 30px;
            width: 66%;
        }
    }
}

.home-icon {
    width: 62px;
}

@media only screen and (max-width: 625px) {
    .guided-answer {
        &__header {
            &__logoAndTitle {
                width: 50%;
            }

            &__sub {
                align-items: start;
                flex-direction: column;
            }

            &__searchField {
                margin-top: 20px;
                align-items: start;
            }
            &__navButtons {
                text-align: center;
                justify-content: center;
            }
            &__navButtons__content {
                margin-left: 0;
            }
            &__navButtons__content__text {
                display: none;
            }
            &__back-restart-buttons {
                justify-content: space-between;
                width: 100%;
            }
            &__allAnswersButton {
                margin-right: 48px;
                width: 28px;
                margin-left: 6px;
            }
        }
    }
    .home-icon {
        width: 28px;
        padding: 0 3px 2px 3px;
        border-radius: 3px;
        color: var(--vscode-menu-foreground, #cccccc);
    }
    .home-icon:hover {
        background: var(--vscode-menubar-selectionBackground);
    }
}
