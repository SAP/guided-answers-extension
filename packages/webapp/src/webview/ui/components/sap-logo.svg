<svg width="53" height="26" viewBox="0 0 53 26" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0 0V26H26.7843L53 0H0ZM32.2961 15.9976H30.6111V20.9663H22.884L22.1174 18.5889C21.2997 18.8549 20.3749 19.0038 19.3975 19.0038C18.42 19.0038 17.4457 18.8454 16.612 18.5667L15.8166 20.9663H11.0795L11.6017 19.5788C11.5394 19.6342 11.4772 19.6897 11.4117 19.7419C10.2298 20.6844 8.75241 21.2039 6.92047 21.2356H6.64896C4.5471 21.2372 2.48677 20.6622 0.744274 19.6802L2.3558 16.506C4.10149 17.5292 5.19555 17.7573 6.67611 17.7288C7.44274 17.7145 7.99536 17.5751 8.35791 17.2045C8.57513 16.978 8.69332 16.6992 8.69811 16.3983C8.71248 15.5731 7.53059 15.1866 6.08356 14.7431C4.90966 14.3804 3.57763 13.8894 2.53788 13.1323C1.30328 12.2326 0.715526 11.108 0.744274 9.51934C0.76344 8.36467 1.19467 7.32403 1.99325 6.50198C3.14001 5.32671 4.99111 4.61237 7.21116 4.61237H7.22393C9.12774 4.61553 11.45 5.14931 13.1334 5.99988L11.3174 9.13128C9.47752 8.24904 8.55117 8.16668 7.50184 8.11124C5.94142 8.02729 5.15083 8.58166 5.14124 9.1788C5.12847 9.90582 6.61542 10.5663 8.02411 11.013C10.1531 11.6846 12.8587 12.5826 13.2708 15.1739L17.1598 4.88638H21.6782L26.2365 17.1871V4.88638H31.4528C36.4391 4.88638 38.7821 6.65087 38.7821 10.4919C38.7821 13.9416 36.3704 15.9976 32.2961 15.9976V15.9976Z" fill= var(--vscode-settings-headerForeground)/>
<path d="M31.7593 8.27124H30.6094V12.4417H31.7593C33.2926 12.4417 34.5128 11.9364 34.5128 10.3319C34.5128 8.72741 33.2926 8.27124 31.7593 8.27124Z" fill= var(--vscode-settings-headerForeground)/>
<path d="M19.3783 9.9502L17.7236 15.1233C18.2156 15.3086 18.7873 15.4163 19.3959 15.4163C20.0044 15.4163 20.5458 15.3133 21.0329 15.1391L19.4134 9.95178H19.3799L19.3783 9.9502Z" fill= var(--vscode-settings-headerForeground)/>
</svg>
