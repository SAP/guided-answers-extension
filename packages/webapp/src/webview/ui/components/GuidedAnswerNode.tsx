import { default as parse, DOMNode, Element, Text } from 'html-react-parser';
import React, { ReactElement } from 'react';
import { GuidedAnswerNode, HTML_ENHANCEMENT_DATA_ATTR_MARKER } from '@sap/guided-answers-extension-types';
import { useSelector } from 'react-redux';
import { actions } from '../../state';
import { AppState } from '../../types';
import './GuidedAnswerNode.scss';
import { GuidedAnswerNavPath } from './GuidedAnswerNavPath';

/**
 * Replacer function for html-react-parser's replace function. If an element was marked, replace it with  link <a>
 * The command JSON is in the data-* attribute.
 *
 * @param domNode - current DOM node from html-react-parser
 * @returns - undefined if nothing to replace; the new node (<a>) in case of replacement
 */
function replace(domNode: DOMNode): JSX.Element | undefined {
    if (domNode.type === 'tag') {
        const domElement: Element = domNode as Element;
        const dataCommandString = domElement?.attribs?.[HTML_ENHANCEMENT_DATA_ATTR_MARKER];
        if (dataCommandString) {
            try {
                const command = JSON.parse(decodeURIComponent(dataCommandString));
                const textContent = domElement?.firstChild?.type === 'text' ? (domElement.firstChild as Text).data : '';
                if (command) {
                    return (
                        <a
                            href="javascript:void(0)"
                            onClick={(): void => {
                                actions.executeCommand(command);
                            }}>
                            {textContent}
                        </a>
                    );
                }
            } catch (error) {
                console.error(error);
            }
        }
    }
}

/**
 * Process the enhancements in an HTML string, which includes parsing the HTML and replacing the respective nodes.
 *
 * @param htmlString - HTML string
 * @returns - react elements including enhancements
 */
function enhanceBodyHtml(htmlString: string): ReactElement | undefined {
    let result;
    try {
        result = parse(htmlString, { replace });
    } catch (error) {
        console.error(error);
    }
    return result as ReactElement;
}

/**
 * Check if an HTML string contains an enhancement.
 *
 * @param htmlString - HTML string
 * @returns true: has enhancementsl; false: no enhancements
 */
function hasEnhancements(htmlString: string): boolean {
    return typeof htmlString === 'string' && htmlString.includes(HTML_ENHANCEMENT_DATA_ATTR_MARKER);
}

/**
 * Return the navigation section.
 *
 * @returns - react elements for navigation
 */
function getNavigationSection(): ReactElement {
    return (
        <div id="left" className="column">
            <GuidedAnswerNavPath />
        </div>
    );
}

/**
 * Return the content for a Guided Answers node, which consists of the middle part, the actual Guided Answers text
 * and, if present, the right part with additional commands.
 *
 * @param activeNode - the active Guided Answers node
 * @returns - react element for content
 */
function getContent(activeNode: GuidedAnswerNode): ReactElement {
    const enhancedBody = hasEnhancements(activeNode.BODY) ? enhanceBodyHtml(activeNode.BODY) : null;
    const middle = (
        <div id="middle" className="column">
            <h1>
                {activeNode.TITLE} <span className="light-text">#{activeNode.NODE_ID}</span>
            </h1>
            {enhancedBody ? enhancedBody : <div dangerouslySetInnerHTML={{ __html: activeNode.BODY }}></div>}
            <p className="guided-answer__node__question">{activeNode.QUESTION}</p>
            <div className="guided-answer__node">
                {activeNode.EDGES.map((edge, index) => (
                    <div
                        key={`edge_button${index}`}
                        className="guided-answer__node__edge"
                        onClick={(): void => {
                            actions.selectNode(edge.TARGET_NODE);
                        }}>
                        {edge.LABEL}
                    </div>
                ))}
            </div>
        </div>
    );
    const right = activeNode.COMMANDS ? (
        <div id="right" className="column">
            <div className="guided-answer__node__commands">
                {activeNode.COMMANDS
                    ? activeNode.COMMANDS.map((command, index) => (
                          <div className="guided-answer__node__command">
                              <div className="guided-answer__node__command__header">
                                  {/* <div className="guided-answer__node__command__header__icon">
                              {command.icon}
                          </div> */}
                                  <div className="guided-answer__node__command__header__label">{command.label}</div>
                              </div>

                              <div
                                  key={`command-${index}`}
                                  className="guided-answer__node__command__description"
                                  onClick={(): void => {
                                      actions.executeCommand(command);
                                  }}>
                                  {command.description}
                              </div>
                          </div>
                      ))
                    : ''}
            </div>
        </div>
    ) : null;

    return right ? (
        <div className="main-container">
            {middle}
            {right}
        </div>
    ) : (
        middle
    );
}

/**
 * Render the react elements to display a Guided Answers node.
 *
 * @returns - react element of Guided Answers node
 */
export function GuidedAnswerNode(): ReactElement {
    const nodes = useSelector<AppState, GuidedAnswerNode[]>((state) => state.activeGuidedAnswerNode);
    const activeNode = nodes[nodes.length - 1];
    return activeNode ? (
        <section className="guided-answer__node__body">
            {getNavigationSection()}
            {getContent(activeNode)}
        </section>
    ) : (
        <></>
    );
}
