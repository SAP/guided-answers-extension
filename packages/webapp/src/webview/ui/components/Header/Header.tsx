import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { AppState } from '../../../types';
import { AllAnswersButton, BackButton, GeneralFeedbackButton, RestartButton } from './NavigationButtons';
import { Logo } from './Logo';
import './Header.scss';
import { SearchField } from './SearchField';
import { FocusZone } from '@fluentui/react-focus';

/**
 * Renders and returns the header section.
 *
 * @param props Props for Header component
 * @param props.showSub Show/Hide subsection of the Header
 * @param props.showNavButons Show/Hide navigation buttons
 * @param props.showLogo Show/Hide SAP logo
 * @param props.showSearch Show/Hide Search field
 * @returns - react element for header section
 */
export function Header(props: {
    showSub: boolean;
    showNavButons: boolean;
    showLogo: boolean;
    showSearch: boolean;
}): ReactElement {
    const appState = useSelector<AppState, AppState>((state) => state);
    return (
        <div className="guided-answer__header" style={{ paddingBottom: props.showNavButons === true ? '0' : '20px' }}>
            {props.showSub === true ? (
                <>
                    <div className="guided-answer__header__sub">
                        {!!props.showLogo && <Logo />}
                        {!!props.showSearch && <SearchField />}
                    </div>{' '}
                </>
            ) : (
                <></>
            )}

            {props.showNavButons === true ? (
                <>
                    <FocusZone isCircularNavigation={true} className="guided-answer__header">
                        <div className="guided-answer__header__allAnswersButton">
                            <AllAnswersButton />
                        </div>

                        <div className="guided-answer__header__back-restart-buttons">
                            <BackButton />
                            <div className="guided-answer__header__restart-feedback-buttons">
                                {appState.activeGuidedAnswerNode.length > 1 && <RestartButton />}
                                {appState.activeGuidedAnswerNode.length > 1 && (
                                    <div className="guided-answer__header__divider"></div>
                                )}
                                <GeneralFeedbackButton />
                            </div>
                        </div>
                    </FocusZone>
                </>
            ) : (
                <></>
            )}
        </div>
    );
}
