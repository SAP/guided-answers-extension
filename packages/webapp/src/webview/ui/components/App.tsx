import type { ReactElement } from 'react';
import React from 'react';
import { useSelector } from 'react-redux';
import { VscSearch } from 'react-icons/vsc';
import { VSCodeButton, VSCodeTextField } from '@vscode/webview-ui-toolkit/react';
import { AppState } from '../../types';
import './App.scss';
import { actions } from '../../state';
import { GuidedAnswerNode } from './GuidedAnswerNode';

import Logo from './sap-logo.svg';

let timer: NodeJS.Timeout;

/**
 * Start element for Guided Answers Extension app.
 *
 * @returns - react elements for this app
 */
export function App(): ReactElement {
    const appState = useSelector<AppState, AppState>((state) => state);

    let content;
    if (appState.activeGuidedAnswerNode.length > 0) {
        content = <GuidedAnswerNode />;
    } else if (appState.guidedAnswerTrees) {
        content = (
            <>
                <VSCodeTextField
                    className="tree-search-field"
                    value={appState.query}
                    placeholder="Search Guided Answers"
                    onInput={(e: any) => {
                        console.log(`Value changed`, e);
                        const newValue = e?.target?.value;
                        if (newValue !== undefined) {
                            clearTimeout(timer);
                            actions.setQueryValue(newValue);
                            timer = setTimeout(() => {
                                actions.searchTree(newValue);
                            }, 400);
                        }
                    }}>
                    <span slot="end" className="tree-search-icon">
                        <VscSearch />
                    </span>
                </VSCodeTextField>
                {appState.guidedAnswerTrees.map((tree, index) => {
                    return (
                        <div key={`tree-item-${index}`} className="tree-item">
                            <div
                                className="guided-answer__tree"
                                onClick={(): void => {
                                    actions.setActiveTree(tree);
                                    actions.selectNode(tree.FIRST_NODE_ID);
                                }}>
                                <ul className="guided-answer__tree__ul">
                                    <h3 className="guided-answer__tree__title">{tree.TITLE}</h3>
                                    {tree.DESCRIPTION && (
                                        <span className="guided-answer__tree__desc">{tree.DESCRIPTION}</span>
                                    )}
                                </ul>
                            </div>
                        </div>
                    );
                })}
            </>
        );
    }
    const backButton =
        appState.activeGuidedAnswerNode.length > 0 ? (
            <VSCodeButton
                className="guided-answer__header__goback"
                onClick={(): void => {
                    actions.goToPreviousPage();
                }}>
                Go back
            </VSCodeButton>
        ) : (
            <></>
        );
    return (
        <div className="guided-answer">
            <div className="guided-answer__header">
                <div className="guided-answer__header__sub">
                    <span id="sap-logo">
                        <Logo />
                    </span>
                    <h1 className="guided-answer__header__title">Guided Answers</h1>
                    <span className="guided-answer__header__subtitle">
                        {appState.activeGuidedAnswer ? ': ' + appState.activeGuidedAnswer.TITLE : ''}
                    </span>
                </div>
                {backButton}
            </div>
            <main className="guided-answer__container">{content}</main>
        </div>
    );
}
