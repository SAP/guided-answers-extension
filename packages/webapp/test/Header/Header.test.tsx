import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../src/webview/ui/components/Header';

jest.mock('@vscode/webview-ui-toolkit/react', () => ({
    VSCodeTextField: () => (
        <>
            <div>SearchField</div>
        </>
    )
}));

jest.mock('react-redux', () => {
    const lib = jest.requireActual('react-redux');
    const state = { activeGuidedAnswerNode: [{ a: 0 }, { b: 0 }] };
    return {
        ...lib,
        useSelector: () => state
    };
});

describe('<Header />', () => {
    let wrapper: any;

    afterEach(() => {
        jest.clearAllMocks();
        wrapper.unmount();
    });

    it('Should render a Header component', () => {
        wrapper = shallow(<Header showSub={true} showLogo={true} showNavButons={false} showSearch={true} />);
        expect(wrapper.find('.guided-answer__header').length).toBe(1);
        expect(wrapper.find('.guided-answer__header__sub').length).toBe(1);
        expect(wrapper.find('Logo').length).toBe(1);
        expect(wrapper.find('SearchField').length).toBe(1);

        wrapper = shallow(<Header showSub={false} showLogo={false} showNavButons={true} showSearch={false} />);
        expect(wrapper.find('.guided-answer__header__allAnswersButton').length).toBe(1);
        expect(wrapper.find('Memo(AllAnswersButton)').length).toBe(1);
        expect(wrapper.find('.guided-answer__header__back-restart-buttons').length).toBe(1);
        expect(wrapper.find('BackButton').length).toBe(1);
        expect(wrapper.find('RestartButton').length).toBe(1);
    });
});
