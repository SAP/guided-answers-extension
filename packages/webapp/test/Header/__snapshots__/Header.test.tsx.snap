// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render a Header component with the navigation buttons 1`] = `
<div>
  <div
    class="guided-answer__header"
    style="padding-bottom: 0px;"
  >
    <div
      class="ms-FocusZone css-115 guided-answer__header"
      data-focuszone-id="FocusZone9"
    >
      <div
        class="guided-answer__header__allAnswersButton"
      >
        <button
          class="guided-answer__header__navButtons home-icon"
          id="all-answers-button"
          tabindex="0"
          title="Home"
        >
          <svg
            class="guided-answer__header__navButtons__content"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.36 1.37l6.36 5.8-.71.71L13 6.964v6.526l-.5.5h-3l-.5-.5v-3.5H7v3.5l-.5.5h-3l-.5-.5V6.972L2 7.88l-.71-.71 6.35-5.8h.72zM4 6.063v6.927h2v-3.5l.5-.5h3l.5.5v3.5h2V6.057L8 2.43 4 6.063z"
              fill-rule="evenodd"
            />
          </svg>
           
          <span
            class="guided-answer__header__navButtons__content__text text-underline"
          >
            Home
          </span>
        </button>
      </div>
      <div
        class="guided-answer__header__back-restart-buttons"
      >
        <button
          class="guided-answer__header__navButtons"
          id="back-button"
          tabindex="-1"
          title="Step back"
        >
          <svg
            class="guided-answer__header__navButtons__content"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7 3.093l-5 5V8.8l5 5 .707-.707-4.146-4.147H14v-1H3.56L7.708 3.8 7 3.093z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="guided-answer__header__navButtons__content text-underline"
            style="margin-left: 3px;"
          >
            Step back
          </span>
        </button>
        <div
          class="guided-answer__header__restart-feedback-buttons"
        >
          <button
            class="guided-answer__header__navButtons"
            id="restart-button"
            tabindex="-1"
            title="Restart"
          >
            <svg
              class="guided-answer__header__navButtons__content"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.681 3H2V2h3.5l.5.5V6H5V4a5 5 0 1 0 4.53-.761l.302-.954A6 6 0 1 1 4.681 3z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="guided-answer__header__navButtons__content text-underline"
              style="margin-left: 3px;"
            >
              Restart
            </span>
          </button>
          <div
            class="guided-answer__header__divider"
          />
          <button
            class="guided-answer__header__navButtons"
            tabindex="-1"
            title="Feedback"
          >
            <i
              aria-hidden="true"
              class="ts-icon guided-answer__header__navButtons__content root-105"
              data-icon-name="ChatBubbles"
              style="padding-top: 2px;"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Header /> Should render a Header component without the navigation buttons 1`] = `
<div>
  <div
    class="guided-answer__header"
    style="padding-bottom: 20px;"
  >
    <div
      class="guided-answer__header__sub"
    >
      <div
        class="guided-answer__header__logoAndTitle"
      >
        <span
          id="sap-logo"
        >
          <svg />
        </span>
        <h1
          class="guided-answer__header__title"
        >
          Guided Answers
        </h1>
      </div>
      <div
        class="guided-answer__header__searchField"
      >
        <div
          class="ms-SearchBox tree-search-field root-109"
        >
          <div
            aria-hidden="true"
            class="ms-SearchBox-iconContainer iconContainer-110"
          >
            <i
              aria-hidden="true"
              class="ms-SearchBox-icon icon-114"
              data-icon-name="Search"
            />
          </div>
          <input
            class="ms-SearchBox-field field-113"
            id="search-field"
            placeholder="Search Guided Answers"
            role="searchbox"
            value=""
          />
        </div>
        <div
          id="filters"
        >
          <div
            class="ms-FocusZone css-115"
            data-focuszone-id="FocusZone0"
            style="display: flex;"
          >
            <button
              aria-disabled="true"
              class="ms-Button ms-Button--icon is-disabled filter-button filter-button-selected root-116"
              data-is-focusable="false"
              disabled=""
              id="filter-products"
              tabindex="-1"
              title="Filter Products"
              type="button"
            >
              <span
                class="ms-Button-flexContainer flexContainer-117"
                data-automationid="splitbuttonprimary"
              >
                <i
                  aria-hidden="true"
                  class="ms-Icon root-105 ms-Button-icon icon-119"
                  data-icon-name="Table"
                />
              </span>
            </button>
            <button
              aria-disabled="true"
              class="ms-Button ms-Button--icon is-disabled filter-button filter-button-selected root-116"
              data-is-focusable="false"
              disabled=""
              id="filter-components"
              tabindex="-1"
              title="Filter Components"
              type="button"
            >
              <span
                class="ms-Button-flexContainer flexContainer-117"
                data-automationid="splitbuttonprimary"
              >
                <i
                  aria-hidden="true"
                  class="ms-Icon root-105 ms-Button-icon icon-119"
                  data-icon-name="IdTag"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
     
  </div>
</div>
`;
